test/test-rewrite-url.js

```diff
@@ -0,0 +1,177 @@
+                  slug: '/tab-two/child-one'
+                },
+                {
+                  title: 'Tab Two Child Two',
+                  type: 'default',
+                  slug: '/tab-two/child-two'
+                },
+              ]
+            },
+            {
+              title: 'Products',
+              type: 'products-page',
+              slug: '/products'
+            }
+          ],
+          types: [
+            {
+              name: 'home',
+              label: 'Home'
+            },
+            {
+              name: 'default',
+              label: 'Default'
+            },
+            {
+              name: 'products',
+              label: 'Products'
+            }
+          ]
+        },
+        'products': {
+          extend: 'apostrophe-pieces',
+          name: 'product'
+        },
+        'products-pages': {
+          extend: 'apostrophe-pieces-pages'
+        },
+      },
+      afterInit: function(callback) {
+        assert(apos.modules['apostrophe-site-map']);
+        return callback(null);
+      },
+      afterListen: function(err) {
+        done();
+      }
+    });
+  });
+
+  it('insert a product for test purposes', function(done) {
+    var product = _.assign(apos.modules.products.newInstance(), {
+      title: 'Cheese',
+      slug: 'cheese'
+    });
+    apos.modules.products.insert(apos.tasks.getReq(), product, function(err) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('make sure everything is published and out of the trash for test purposes', function(done) {
+    return apos.docs.db.update({}, {
+      $set: {
+        trash: false,
+        published: true
+      }
+    }, {
+      multi: true
+    }, function(err, count) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('insert an unpublished product for test purposes', function(done) {
+    var product = _.assign(apos.modules.products.newInstance(), {
+      title: 'Rocks',
+      slug: 'rocks',
+      published: false
+    });
+    apos.modules.products.insert(apos.tasks.getReq(), product, function(err) {
+      assert(!err);
+      done();
+    });
+  });
+
+  it('should generate a suitable sitemap', function(done) {
+    this.timeout(10000);
+    get('http://localhost:7780/sitemap.xml', function(err, xml) {
+      if (err) {
+        console.error(err);
+      }
+      assert(!err);
+      assert(xml);
+      assert(xml.indexOf('<loc>https://public-site.com/</loc>') !== -1);
+      assert(xml.indexOf('<loc>https://public-site.com/tab-one</loc>') !== -1);
+      assert(xml.indexOf('<loc>https://public-site.com/tab-two</loc>') !== -1);
+      assert(xml.indexOf('<loc>https://public-site.com/tab-one/child-one</loc>') !== -1);
+      assert(xml.indexOf('<loc>https://public-site.com/products/cheese</loc>') !== -1);
+      assert(xml.indexOf('<loc>https://public-site.com/products/rocks</loc>') === -1);
+      done();
+    });
+  });
+
+});
+
+function get(url, callback) {
```

Comment:
You can use `mocha` with promises, example:

```javascript
  it('should generate a suitable sitemap', async () => {
    try {
      const xml = await get('http://localhost:7780/sitemap.xml');

      assert(xml);
      assert(xml.includes('<loc>https://public-site.com/</loc>'));
      assert(xml.includes('<loc>https://public-site.com/tab-one</loc>'));
      assert(xml.includes('<loc>https://public-site.com/tab-two</loc>'));
      assert(xml.includes('<loc>https://public-site.com/tab-one/child-one</loc>'));
      assert(xml.includes('<loc>https://public-site.com/products/cheese</loc>'));
      assert(!xml.includes('<loc>https://public-site.com/products/rocks</loc>'));
    } catch (err) {
      assert(!err);
      console.error(err);
    }
  });
```

And modify the get method to return a promise, this way:
```javascript
function get(url) {
  return new Promise((resolve, reject) => {
    request(url, function(err, response, body) {
      if (err || (response.statusCode >= 400)) {
        reject(err || response.statusCode);
      }
      resolve(body);
    });
  })
}
```